### ISSUE TITLE
No notifications

### ISSUE THREAD

**Issue by PapsOu** [dino/dino#429]
Hello,

I'm using dino (commit bf2c78d1b720fff381b2a5a86bb154cffc0d7814) under fedora 28 (kernel 4.18.5-200.fc28.x86_64) and GNOME Shell 3.28.3.

I have no notifications on incoming messages even when setting the notification parameter to `all`

I've built dino with the following flag `./configure --enable-plugin=notification-sound` but there is no notification (no sound and no notification popup). There is no notification popup without the flag `--enable-plugin=notification-sound`.

Debugging with the `--print-xmpp=all` does not help at all.

What can I do to help you solving this issue ?

**Comment by licaon-kter** [comment]
I have `dunst` installed that shows them (no sound), but I build without any `enable`

Other apps show notifications ok?

**Comment by PapsOu** [comment]
Hello @licaon-kter, other applications show notifications popup (rhytmbox, pidgin, etc.). I'll try installing dunst soon.

**Comment by PapsOu** [comment]
Well, replacing gnome default dbus notification with dunst seems to be not so easy.

triggering `notify-send test` displays a notification on the desktop. Does dino can use notify-send if present on the OS ? (Like this example using `Posix` library : `Posix.system("notify-send ...")` )

**Comment by licaon-kter** [comment]
Just starting dunst doesn't cut it?

**Comment by PapsOu** [comment]
No, starting `dunst` prints this : 

```bash
Name Lost. Is Another notification daemon running?
```

And I don't know which is the current one.

**Comment by PapsOu** [comment]
I've found where the notification is not sent: https://github.com/dino/dino/blob/master/libdino/src/service/notification_events.vala#L47

Commenting out the return https://github.com/dino/dino/blob/master/libdino/src/service/notification_events.vala#L52 seems to solve the problems with notifications

I've made a PR (https://github.com/dino/dino/pull/432) for this but I didn't check what this change can break.

**Comment by fiaxh** [comment]
Thank you for reporting and investigating.

Dino does not notify new messages until MAM catchup is done, so that it does not unnecessarily notify messages for which a read-marker is already stored in (and will be received from) MAM. Your PR would break that. However if the server does not support MAM, Dino would likely produce the behavior you observe.

Does your server offer MAM? You can check by running dino as `./dino --print-xmpp="iq[type=yourusername@yourdomain.you].{http://jabber.org/protocol/disco#info}:query")` (replace `yourusername@yourdomain.you`) and checking for an entry `<feature var='urn:xmpp:mam:...' />`.

**Comment by PapsOu** [comment]
The debug query display nothing.

To be sure, I've done a `--print-xmpp=all` and searched into output the part `feature var='urn:xmpp:mam` without any results. (I've found many `feature var='urn:xmpp:` but nothing with `mam`, even searching only the string `mam` case insensitive)

I'm using a GTalk account and an XMPP company account.

**Comment by licaon-kter** [comment]
Use https://github.com/iNPUTmice/caas to get your server features

**Comment by PapsOu** [comment]
Thanks @licaon-kter, your online tool confirmes that the server does not handle XEP-0313: Message Archive Management

**Comment by PapsOu** [comment]
Maybe using something like this https://github.com/dino/dino/blob/master/libdino/src/service/conversation_manager.vala#L137 to check if the current conversation supports MAM and avoid returning if MAM is not supported ?

**Comment by qazip** [comment]
@licaon-kter, can you show me your dunstrc please? I also have dunst installed (with i3wm) and I do not get any notifications from dino.

**Comment by licaon-kter** [comment]
@qazip  here it is:
```
[global]
    font = Terminus
    format = "%s %b"
    sort = yes
    indicate_hidden = yes
    alignment = left
    bounce_freq = 0
    show_age_threshold = 60;
    word_wrap = yes
    geometry = "0x10-30+20"
    transparency = 0
    idle_threshold = 120
    monitor = 0
    follow = keyboard
    sticky_history = yes
    line_height = 0
    separator_height = 2;
    separator_color = auto

[shortcuts]
    close = ctrl+shift+grave
    close_all = 
    history = ctrl+grave

[urgency_low]
    background = "#aaaaff"
    foreground = "#000000"
    timeout = 10

[urgency_normal]
    background = "#1793D1"
    foreground = "#DDDDDD"
    timeout = 10

[urgency_critical]
    background = "#ffaaaa"
    foreground = "#000000"
    timeout = 0
```

**Comment by fiaxh** [comment]
The issue was fixed by #432.
